### PR TITLE
chore: remove the `as_field` and `from_field` built-ins

### LINKED ISSUE #6838: Replace the `as_field` and `from_field` built-ins with the `From<Field>` and `Into<Field>` traits
### Problem

A [comment](https://github.com/noir-lang/noir/pull/6469#issuecomment-2548553868) by Jake:

> @asterite only the various integer types. It's an old builtin from before we had traits. We really should remove it and replace it with `From<Field>` and `Into<Field>`

### Happy Case

The built-ins and corresponding intrinsics are replaced and removed.

### Workaround

None

### Workaround Description

_No response_

### Additional Context

_No response_

### Project Impact

None

### Blocker Context

_No response_

### Would you like to submit a PR for this Issue?

None

### Support Needs

_No response_

### PR DESCRIPTION
**Pull request by asterite**
# Description

## Problem

Resolves #6838

## Documentation

Check one:
- [ ] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: noir_stdlib/src/lib.nr [hunk 1]
```diff
@@ -89,28 +89,31 @@ pub fn assert_constant<T>(x: T) {}
...
+        x.into() + 340282366920938463463374607431768211456 - y.into(),
     )
 }
 
-pub fn wrapping_mul<T>(x: T, y: T) -> T {
-    crate::from_field(crate::as_field(x) * crate::as_field(y))
+pub fn wrapping_mul<T>(x: T, y: T) -> T
+where
+    T: convert::Into<Field>,
+    T: convert::From<Field>,
```
Comment (TomAFrench) [inline noir_stdlib/src/lib.nr:114]:
```suggestion
    T: convert::Into<Field> + convert::From<Field>,
```

Is this not possible?

File: noir_stdlib/src/convert.nr [hunk 2]
```diff
@@ -116,4 +116,13 @@ impl From<bool> for Field {
         value as Field
     }
 }
+
+// Field
+impl From<Field> for u64 {
+    fn from(value: Field) -> u64 {
+        value.assert_max_bit_size::<64>();
```
[the remark this answers is not in the document]
Comment (asterite) [reply noir_stdlib/src/convert.nr:123]: Good point. I guess we could introduce a private trait just for this? Maybe `FromFieldUnsafe`.

[the remark this answers is not in the document]
Comment (jfecher) [reply noir_stdlib/src/convert.nr:123]: It is worth noting that Rust has the same rule but also implements this

[the remark this answers is not in the document]
Comment (jfecher) [reply noir_stdlib/src/convert.nr:123]: If we want to keep the check my suggestion would be a `TryFrom` crate like rust. Then `x.try_into().unwrap()`

[the remark this answers is not in the document]
Comment (asterite) [reply noir_stdlib/src/convert.nr:123]: But the original code truncated the result. I see these are used in `wrapping_add`, `wrapping_sub`, etc., so maybe they should never fail.

**Comment by asterite** [comment]
Aztec-Packages is failing now, but I think this uncovered a bug. There's code that does something like this:

```noir
fn main() {
    let u = U128::from_integer(184467440737095516152);
    let y = U128::from_integer(u);
    println(y);
}
```

That is, it's calling `U128::from_integer(x)` with `x` being `U128` itself. That seems to work in `master` but just truncates the U128 value. Just using `x` here is fine.

**Comment by TomAFrench** [comment]
Looks good to me with a private trait (although in reality it's not really private until we require traits to be in scope in order to use its methods).

**Comment by asterite** [comment]
Rust has, in a crate, an [AsPrimitive](https://docs.rs/num-traits/latest/num_traits/cast/trait.AsPrimitive.html) trait that's essentially the `as` keyword but in a programmatic way. In the end that's what `as_field` and `from_field` end up doing, and this PR so far too. I just tried introducing such trait and using it, and it works fine. The benefits are that it's just one trait, it's a more general trait, and it could also be a public trait. What do you think?

**Review by TomAFrench — APPROVED** [review]

**Pull request merged**